src/oiiotool/oiiotool.cpp

```diff
@@ -3458,14 +3458,30 @@ OIIOTOOL_OP(warp, 1, [&](OiiotoolOp& op, span<ImageBuf*> img) {
 
 // --demosaic
 OIIOTOOL_OP(demosaic, 1, [&](OiiotoolOp& op, span<ImageBuf*> img) {
-    std::string pattern   = op.options().get_string("pattern");
-    std::string algorithm = op.options().get_string("algorithm");
-    std::string layout    = op.options().get_string("layout");
-
-    return ImageBufAlgo::demosaic(*img[0], *img[1],
-                                  { { "pattern", pattern },
-                                    { "algorithm", algorithm },
-                                    { "layout", layout } });
+    ParamValueList list;
+    const std::vector<std::string> keys = { "pattern", "algorithm", "layout" };
+    for (const auto& key : keys) {
+        auto iter = op.options().find(key);
+        if (iter != op.options().cend()) {
+            list.push_back(iter[0]);
+        }
+    }
+
+    std::string wb = op.options()["white_balance"];
+    string_view str(wb);
+    float white_balance[4] = { 1.0f, 1.0f, 1.0f, 1.0f };
+    if (str.size() && Strutil::parse_float(str, white_balance[0])
+        && Strutil::parse_char(str, ',')
+        && Strutil::parse_float(str, white_balance[1])
+        && Strutil::parse_char(str, ',')
+        && Strutil::parse_float(str, white_balance[2])
+        && Strutil::parse_char(str, ',')
+        && Strutil::parse_float(str, white_balance[3])) {
```

Comment:
I think there is a more compact alternative, and also more correct, because it will fail if the string contains any extra non-whitespace after the 4th value.


```suggestion
    string_view str(wb);
    Strutil::remove_trailing_whitespace(str);
    float white_balance[4] = { 1.0f, 1.0f, 1.0f, 1.0f };
    if (Strutil::parse_values(str, "", white_balance, ",") && str.size()) {
```

Reply:
Done! I've also decided to allow 3 values, for RGB, and extend them to RGGB automatically.